src/movies.js

```diff
@@ -1,16 +1,179 @@
+    return avg
+}
+
 // Iteration 4: Drama movies - Get the average of Drama Movies
 
+function dramaMoviesRate(movies) {
+    if (movies.length === 0) {
+        return 0
+    }
+    let numOfMovies = 0
+    let avgTot = movies.reduce((tot, e) => {
+        if (e.genre.includes("Drama")) {
+            numOfMovies += 1
+            return tot + e.rate
+        } else {
+            return tot
+        }
+    }, 0)
+    if (numOfMovies === 0) {
+        return 0
+    }
+    let avg = avgTot / numOfMovies
+
+    avg = avg.toFixed(2)
+    avg = Number(avg)
+
+    return avg
+}
+
+
 // Iteration 5: Ordering by year - Order by year, ascending (in growing order)
 
+function orderByYear(movies) {
+    if (movies.length === 0) {
+        return 0
+    }
+    let copyMovie = JSON.parse(JSON.stringify(movies))
+    let sortedMovies = copyMovie.sort((a, b) => {
+        if (a.year > b.year) {
+            return 1
+        } else if (a.year < b.year) {
+            return -1
+        } else {
+            if (a.title > b.title) {
+                return 1
+            } else if (a.title < b.title) {
+                return -1
+            } else {
+
+                return 0
+            }
+            return 0
+        }
+    })
+    return sortedMovies
+}
+
 // Iteration 6: Alphabetic Order - Order by title and print the first 20 titles
 
+function orderAlphabetically(movies) {
+    let copyMovies = JSON.parse(JSON.stringify(movies))
+    let ordByAlph = copyMovies.sort((a, b) => {
+        if (a.title > b.title) {
+            return 1
+        } else if (a.title < b.title) {
+            return -1
+        } else {
+
+            return 0
+        }
+    }).map((i) => {
+        return i.title
+    })
+
+    return ordByAlph.splice(0, 20)
+}
+
+
 // BONUS - Iteration 7: Time Format - Turn duration of the movies from hours to minutes
 
+function turnHoursToMinutes(movies) {
+    let copyMovies = JSON.parse(JSON.stringify(movies))
+    let hourToMinutes = copyMovies.map((i) => {
+        if (i.duration.includes('h') && i.duration.includes('min') && Number(i.duration[4]) == Number) {
+            let minutes = i.duration.slice(0, 1) * 60 + Number(i.duration.slice(3, 4))
+            return i.duration = minutes
+        } else if (i.duration.includes('h') && i.duration.includes('min')) {
+            let minutes = i.duration.slice(0, 1) * 60 + Number(i.duration.slice(3, 5))
+            return i.duration = minutes
+        } else if (i.duration.includes('h')) {
+            let minutes = i.duration.slice(0, 1) * 60
+            return i.duration = minutes
+        } else if (i.duration.includes('min') && Number(i.duration[1]) == Number) {
+            let minutes = i.duration.slice(0, 3)
+            return i.duration = minutes
+        } else if (i.duration.includes('min')) {
+            let minutes = Number(i.duration.slice(0, 2))
+            return i.duration = minutes
+        } else {
+            return 0
+        }
+    })
+    return copyMovies
```

Comment:
Very good solution! There might be a way to reduce a little bit the amount of conditions (for example merge 1 and 2, 4 and 5) However, still very impressive. Keep it up! :)